changes to errgroup.go break 32 bit arm

Sorry for this, i'm planning to migrate to 64bit machines.

<!--- Provide a general summary of the issue in the Title above -->

## Expected Behavior
<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->

minio starts succecfully `after mc admin update`


## Current Behavior
<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from current behavior -->

Minio fails with:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x11a20]
goroutine 137 [running]:
runtime/internal/atomic.goCas64(0x3b5091c, 0xffffffff, 0xffffffff, 0x3, 0x0, 0x0)
runtime/internal/atomic/atomic_arm.go:87 +0x1c
github.com/minio/minio/pkg/sync/errgroup.(*Group).Go.func1(0x3b50900, 0x3, 0x3b509c0)
github.com/minio/minio/pkg/sync/errgroup/errgroup.go:124 +0x11c
created by github.com/minio/minio/pkg/sync/errgroup.(*Group).Go
         github.com/minio/minio/pkg/sync/errgroup/errgroup.go:104 +0x5
```

## Possible Solution
<!--- Not obligatory, but suggest a fix/reason for the bug, -->
<!--- or ideas how to implement the addition or change -->

64 bit align the Group  struct.
firstErr is the only fields that is accessed atomically so ist should be enough to move this field to the beginning of the struct.

> On ARM, x86-32, and 32-bit MIPS, it is the caller's responsibility to arrange for 64-bit alignment of 64-bit words accessed atomically. The first word in a variable or in an allocated struct, array, or slice can be relied upon to be 64-bit aligned.


## Regression
<!-- Is this issue a regression? (Yes / No) -->
<!-- If Yes, optionally please include minio version or commit id or PR# that caused this regression, if you have these details. -->
Yes

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* Version used (`minio --version`): RELEASE.2021-02-07T01-31-02Z
* Server setup and configuration: Distributed, 4 Servers with one drive each
* Operating System and version (`uname -a`): Linux odroidxu4 4.14.212-odroidxu4 #12 SMP PREEMPT Wed Feb 3 21:31:20 CET 2021 armv7l armv7l armv7l GNU/Linux

--- a/cmd/rest/client.go
+++ b/cmd/rest/client.go
@@ -60,6 +60,8 @@ func (n *NetworkError) Unwrap() error {
 
 // Client - http based RPC client.
 type Client struct {
+	connected int32 // ref: https://golang.org/pkg/sync/atomic/#pkg-note-BUG
+
 	// HealthCheckFn is the function set to test for health.
 	// If not set the client will not keep track of health.
 	// Calling this returns true or false if the target
@@ -84,7 +86,6 @@ type Client struct {
 	httpClient   *http.Client
 	url          *url.URL
 	newAuthToken func(audience string) string
-	connected    int32
 }
 
 // URL query separator constants

--- a/pkg/sync/errgroup/errgroup.go
+++ b/pkg/sync/errgroup/errgroup.go
@@ -27,10 +27,10 @@ import (
 //
 // A zero Group can be used if errors should not be tracked.
 type Group struct {
+	firstErr  int64 // ref: https://golang.org/pkg/sync/atomic/#pkg-note-BUG
 	wg        sync.WaitGroup
 	bucket    chan struct{}
 	errs      []error
-	firstErr  int64
 	cancel    context.CancelFunc
 	ctxCancel <-chan struct{} // nil if no context.
 	ctxErr    func() error

